src/main/java/teammates/logic/core/CoursesLogic.java

```diff
@@ -782,4 +792,20 @@ public boolean isCourseArchived(CourseAttributes course, InstructorAttributes in
         }
         return result;
     }
+    
+    public List<String> getArchivedCourseIds(List<CourseDetailsBundle> allCourses, List<InstructorAttributes> instructorList) {
+        List<String> activeCourseIds = new ArrayList<String>();
+        for (CourseDetailsBundle cdb : allCourses) {
+            if (cdb.course.isArchived) {
```

Comment:
I think `isArchived` of the course Object is legacy. We should use `isArchived` from instructor instead